## Aim

Become more familiar with orthology, phylogenies, and comparative genomics.

## Learning outcomes

* Define orthology, paralogy and their subtypes
* Retrieve orthology information from the OMA database
* Map sequences quickly to their Hierarchical Orthologous Groups
* Infer orthologs on custom genomes using the FastOMA pipeline
* Construct and interpret phylogenetic species trees using OMA data

## Exercises

The exercises were implemented as modules in the [OMA Academy](https://oma-stage.vital-it.ch/oma/academy/). OMA Academy is an e-learning website with a suite of self-learning modules centered around comparative genomics and phylogenetics. The modules are specifically designed to help computational biologists use OMA, a method and database for inferring orthologs. Each module is stand-alone, with a basic introduction and a series of exercises with hints and answers.

Here are the links to directly access the different modules:

* [Module 1](https://oma-stage.vital-it.ch/oma/academy/module/OMA_browser_2023)
* [Module 2](https://oma-stage.vital-it.ch/oma/academy/module/OMAmer_2023)
* [Module 3](https://oma-stage.vital-it.ch/oma/academy/module/fastOMA_2023)
* [Module 4](https://oma-stage.vital-it.ch/oma/academy/module/species_tree_2023)
