## Software

These links are provided for information only, they are not needed now for the practical but if you finish any exercises early you may want to investigate the tools/resources a little further by checking out their websites, user guides, and/or their main publications.

**OrthoDB - orthology database**

* [Website](https://www.orthodb.org/)
* [Publication](https://academic.oup.com/nar/article/49/D1/D389/5983625)
* [Userguide](https://www.ezlab.org/orthodb_userguide.html)

**BUSCO - Benchmarking Universal Single-Copy Orthologues**

* [Website](https://busco.ezlab.org/)
* [Publication](https://academic.oup.com/mbe/article/38/10/4647/6329644)
* [Userguide](https://busco.ezlab.org/busco_userguide.html)

**MAFFT - multiple sequence alignment tool**

* [Website](https://mafft.cbrc.jp/alignment/software/)
* [Publication](http://mbe.oxfordjournals.org/content/30/4/772)
* [Userguide](https://mafft.cbrc.jp/alignment/software/manual/manual.html)

**TrimAl - alignment filtering tool**

* [Website](http://trimal.cgenomics.org/trimal)
* [Publication](https://academic.oup.com/bioinformatics/article/25/15/1972/213148)
* [Userguide](http://trimal.cgenomics.org/use_of_the_command_line_trimal_v1.2)

**RAxML - maximum likelihood phylogenies**

* [Website](https://cme.h-its.org/exelixis/web/software/raxml/)
* [Publication](https://academic.oup.com/bioinformatics/article/22/21/2688/251208)
* [Userguide](https://cme.h-its.org/exelixis/resource/download/NewManual.pdf)

**Treerecs - tree reconciliation software**

* [Website](https://project.inria.fr/treerecs/)
* [Publication](https://project.inria.fr/treerecs/publications/)
* [Userguide](https://project.inria.fr/treerecs/treerecs-options/)

**CAFE - computational analysis of gene family evolution**

* [Website](https://github.com/hahnlab/CAFE5)
* [Publication](https://academic.oup.com/bioinformatics/article/36/22-23/5516/6039105)
* [Userguide](https://github.com/hahnlab/CAFE5)

**iTOL - online phylogenetic tree viewer**

* [Website](https://itol.embl.de/)
* [Publication](https://academic.oup.com/nar/article/49/W1/W293/6246398)
* [Userguide](https://itol.embl.de/help.cgi)

**Chronos function of the ape package in R**

* [Website/Userguide](https://rdrr.io/cran/ape/man/chronos.html)
* [Publication](https://www.sciencedirect.com/science/article/pii/S1055790313000651)

**JalView: for multiple sequence alignment editing, visualisation and analysis**

* [Website](https://www.jalview.org/)
* [Publication](https://dx.doi.org/10.1093/bioinformatics/btp033)

## Folder with intermediate result files

If you get stuck, you can find all intermediate results files in the virtual gitpod workspace `/workspace/biodivinfo/data/`. Simply fetch the correct output files in order to continue with the next steps of the practical:

* Use the copy `cp` command from the terminal in gitpod, _e.g._:<br>
`cp /workspace/biodivinfo/data/the_file_you_want_to_copy path/to/destination/`

* If the files are zipped (_i.e._ if the name ends with `.zip`), you can uncompress them with:<br>
`unzip /path/to/file.zip`
